src/network/mcpe/handler/ResourcePacksPacketHandler.php

```diff
@@ -65,9 +65,19 @@ public function __construct(
 	){}
 
 	public function setUp() : void{
-		$resourcePackEntries = array_map(static function(ResourcePack $pack) : ResourcePackInfoEntry{
+		$resourcePackEntries = array_map(function(ResourcePack $pack) : ResourcePackInfoEntry{
 			//TODO: more stuff
-			return new ResourcePackInfoEntry($pack->getPackId(), $pack->getPackVersion(), $pack->getPackSize(), "", "", "", false);
+			$encryptionKey = $this->resourcePackManager->getPackEncryptionKey($pack->getPackId());
+
+			return new ResourcePackInfoEntry(
+				$pack->getPackId(),
+				$pack->getPackVersion(),
+				$pack->getPackSize(),
+				$encryptionKey ?? "",
+				"",
+				$encryptionKey === null ? "" : $pack->getPackId(), // ContentId is required for client-side validations when the pack is encrypted.
```

Comment:
Is there any reason not to just always specify this? I don't actually remember what it's for, but there's a similar Geyser PR for reference: https://github.com/GeyserMC/Geyser/pull/3263

Reply:
I just don't think it makes much sense to special-case it, if it works for both encrypted and unencrypted.